examples/Remarketing/AddFlightsFeed.php

```diff
@@ -0,0 +1,423 @@
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())
+            ->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        // Creates a new flights feed.
+        $feedResourceName = self::createFeed($googleAdsClient, $customerId);
+
+        // Get the newly creates feed's attributes and packages them into a map. This read operation
+        // is required to retrieve the attribute IDs.
+        $placeHoldersToFeedAttributesMap =
+            self::getFeed($googleAdsClient, $customerId, $feedResourceName);
+        // Creates the feed mapping.
+        self::createFeedMapping(
+            $googleAdsClient,
+            $customerId,
+            $feedResourceName,
+            $placeHoldersToFeedAttributesMap
+        );
+        // Creates a feed item.
+        self::createFeedItem(
+            $googleAdsClient,
+            $customerId,
+            $feedResourceName,
+            $placeHoldersToFeedAttributesMap
+        );
+    }
+
+    /**
+     * Creates a flight feed.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @return string the feed resource name
+     */
+    private static function createFeed(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        // Creates a flight description attribute.
+        $flightDescriptionAttribute = new FeedAttribute([
```

Comment:
All these attribute creations could be factorized into a "createAttribute" method directly called  from the feed object creation

Reply:
Sorry, don't get what you mean.

Reply:
Yes this is what I meant and I agree with your points.